src/backend/distributed/planner/multi_planner.c

```diff
@@ -294,176 +333,235 @@ CreateDistributedPlan(PlannedStmt *localPlan, Query *originalQuery, Query *query
 
 
 /*
- * GetMultiPlan returns the associated MultiPlan for a PlannedStmt if the
- * statement requires distributed execution, NULL otherwise.
+ * GetMultiPlan returns the associated MultiPlan for a CustomScan.
  */
 MultiPlan *
-GetMultiPlan(PlannedStmt *result)
+GetMultiPlan(CustomScan *customScan)
 {
-	char *serializedMultiPlan = NULL;
 	MultiPlan *multiPlan = NULL;
 
-	serializedMultiPlan = GetMultiPlanString(result);
-	multiPlan = (MultiPlan *) CitusStringToNode(serializedMultiPlan);
-	Assert(CitusIsA(multiPlan, MultiPlan));
-
-	return multiPlan;
-}
+	Assert(IsA(customScan, CustomScan));
```

Comment:
Not really, we cast stuff around a fair bit around this. C's signatures aren't exactly enforced ;)